python/lsst/analysis/tools/atools/diaSourceMetrics.py

```diff
@@ -30,26 +30,49 @@
 from lsst.pex.config import Field
 
 from ..actions.scalar import CountAction, DivideScalar
-from ..actions.vector import FlagSelector, GoodDiaSourceSelector
+from ..actions.vector import DownselectVector, FlagSelector, GoodDiaSourceSelector
+from ..contexts import DrpContext
 from ..interfaces import AnalysisTool
 
 
-class NumDiaSourcesMetric(AnalysisTool):
+class NumGoodDiaSourcesMetrics(AnalysisTool):
     """Calculate the number of DIA Sources that do not have known
-    bad/quality flags set to true.
+    bad/quality flags set to true, and also calculate the ratio of
+    counts of non-flagged sources to all sources.
     """
 
+    parameterizedBand: bool = False
+
+    applyDrpContext = Field[bool](doc="Apply the DRP context if analysing DRP DIA tables.", default=True)
```

Comment:
I think this and https://github.com/lsst/analysis_tools/blob/main/python/lsst/analysis/tools/actions/vector/selectors.py#L415 ... and all mentions of "DrpContext" throughout analysis_tools ...! can all go away now, because all the flag columns are simply named `pixelFlags_thing` and not `base_pixelFlags_flag_thing`.

Reply:
Once @erinleighh has pushed their changes to `GoodDiaSourceSelector`, I'll rebase and remove mention of DrpContext.